### PR TITLE
Header styling conditional render

### PR DESCRIPTION
**Pull request by nickstaylor**
# PR Description

Please include a summary of the change:

This PR styling the header with new NavLink options to appear based on conditional rendering.  Also, new redux variables are introduced in order to account for user navigation of the site.  Minor styling changes appear on some components, major styling changes on the header component.

If it is a new feature, describe the feature and the iteration of the project spec it addresses.

Styles the header and provides links to different pages based on conditional logic.

Note any questions or struggles as well.
In order to include conditional logic, new variables were added to the redux store and brought into the header.  Also, a helper function in the index.js was created (called "appReducer") in order to be able to clear the store when the user clicks on "new address."  Naming changes may be needed in the future for clarity

## Type of change
Please unckeck options that are not relevant.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Add Testing
- [x] Refactor

## Checklist:
- [x] My code follows the Turing style guidelines
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/containers/Charts/Chart.js [hunk 1]
```diff
@@ -162,14 +163,18 @@ const Chart = ({ solarData }) => {
         </div>
       </div>
       <div className="savings-display-bar">
-        <p>Based on your annual energy usage, this solar system would offset {savingsValue}% per year! </p>
+        {history ?
```
Comment (J-Poulter) [inline src/containers/Charts/Chart.js:166]: I like this setup here, that way the user knows what they're missing out on by skipping that previous form

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+  "highlightwords.colors": null,
+  "highlightwords.defaultMode": null,
+  "highlightwords.box": true
+}
```
Comment (alex-latham) [inline .vscode/settings.json:5]: Might want to add .vscode to the .gitignore file. Deleting from PR.

File: src/containers/Configure/Configure.js [hunk 3]
```diff
@@ -35,6 +35,10 @@ const Configure = ({ userQuote, allUserQuotes, loadSolarData }) => {
     if (arrayType === undefined) {
       return updateError('Please select an array type')
     }
+    if (systemSize < 1 || systemSize > 10 || systemLosses < 0 || systemLosses > 20
```
Comment (J-Poulter) [inline src/containers/Configure/Configure.js:38]: Good catch on the input ranges that we need to restrict the users to.

File: src/containers/Configure/Configure.js [hunk 4]
```diff
@@ -35,6 +35,10 @@ const Configure = ({ userQuote, allUserQuotes, loadSolarData }) => {
...
+      || tilt < 0 || tilt > 45 || azimuth < 0 || azimuth > 315) {
+      return updateError('Please reverify inputs to be in acceptable range')
+      }
```
Comment (alex-latham) [inline src/containers/Configure/Configure.js:41]: Nice error handling here

**Pull request merged**